utils/db.py

```diff
@@ -92,12 +93,14 @@ class Db(Mapping):
         tables, like the mapping interface or :py:meth:`values`.
 
     """
-    def __init__(self, hostname=None, credentials=None):
+    def __init__(self, hostname=None, credentials=None, db_port=None):
         self._table_cache = {}
-        if hostname is None:
+        if hostname is None and db_port is None:
```

Comment:
Hmm, yes, the tighter relationship between IPAppliance and Db could be possible, since our Db object is crafted specifically for CFME appliances, is that right?